internal/interpreter/highlight.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+//! This module contains the code for the highlight of some elements
+
+use crate::dynamic_component::{ComponentBox, DynamicComponentVrc};
+use crate::Value;
+use i_slint_compiler::expression_tree::{Expression, Unit};
+use i_slint_compiler::langtype::{ElementType, Type};
+use i_slint_compiler::namedreference::NamedReference;
+use i_slint_compiler::object_tree::{
+    BindingsMap, Component, Document, Element, ElementRc, PropertyAnalysis, PropertyDeclaration,
+    PropertyVisibility, RepeatedElementInfo,
+};
+use i_slint_core::items::ItemRc;
+use i_slint_core::model::{ModelRc, VecModel};
+use std::cell::RefCell;
+use std::path::PathBuf;
+use std::rc::Rc;
+
+const HIGHLIGHT_PROP: &str = "$highlights";
+
+pub fn highlight(component_instance: &DynamicComponentVrc, path: PathBuf, offset: u32) {
+    generativity::make_guard!(guard);
+    let c = component_instance.unerase(guard);
+    let elements = find_element(&c.description().original, path, offset);
+    if elements.is_empty() {
+        c.description()
+            .set_property(c.borrow(), HIGHLIGHT_PROP, Value::Model(ModelRc::default()))
+            .unwrap();
+        return;
+    };
+
+    let elements = elements.into_iter().map(|e| Rc::downgrade(&e)).collect::<Vec<_>>();
+
+    let component_instance = vtable::VRc::downgrade(component_instance);
+    let binding = move || {
+        let component_instance = component_instance.upgrade().unwrap();
+        generativity::make_guard!(guard);
+        let c = component_instance.unerase(guard);
+        let mut values = Vec::<Value>::new();
+        for element in elements.iter().filter_map(|e| e.upgrade()) {
+            if let Some(repeater_path) = repeater_path(&element) {
+                fill_model(&repeater_path, &element, &c, &mut values);
+            }
```

Comment:
If not in a repeater, repeater_path will return Some(vec![]).   It will only return None if the item is not accessible (eg, PopupMenu)
Maybe it would be clearer if y use `Result<Vec<_>, ()>`  instead of Option?

Reply:
Ahh no, I missed the `vec![]` bit - it's all good :)